addons/base_import/models/base_import.py

```diff
@@ -857,33 +1088,34 @@ def _import_image_by_url(self, url, session, field, line_number):
             response.raise_for_status()
 
             if response.headers.get('Content-Length') and int(response.headers['Content-Length']) > maxsize:
-                raise ValueError(_("File size exceeds configured maximum (%s bytes)", maxsize))
+                raise ImportValidationException(_("File size exceeds configured maximum (%s bytes)", maxsize), field=field)
 
             content = bytearray()
             for chunk in response.iter_content(DEFAULT_IMAGE_CHUNK_SIZE):
                 content += chunk
                 if len(content) > maxsize:
-                    raise ValueError(_("File size exceeds configured maximum (%s bytes)", maxsize))
+                    raise ImportValidationException(_("File size exceeds configured maximum (%s bytes)", maxsize), field=field)
 
             image = Image.open(io.BytesIO(content))
             w, h = image.size
             if w * h > 42e6:  # Nokia Lumia 1020 photo resolution
-                raise ValueError(
-                    u"Image size excessive, imported images must be smaller "
-                    u"than 42 million pixel")
+                raise ImportValidationException(_("Image size excessive, imported images must be smaller than 42 million pixel"), field=field)
 
             return base64.b64encode(content)
         except Exception as e:
             _logger.exception(e)
-            raise ValueError(_("Could not retrieve URL: %(url)s [%(field_name)s: L%(line_number)d]: %(error)s") % {
-                'url': url,
-                'field_name': field,
-                'line_number': line_number + 1,
-                'error': e
-            })
-
-    def do(self, fields, columns, options, dryrun=False):
-        """ Actual execution of the import
+            raise ImportValidationException(
+                _("Could not retrieve URL: %(url)s [%(field_name)s: L%(line_number)d]: %(error)s") % {
```

Comment:
As our dear friend Martin explained, now translations work by passing arguments into the _() function.
To re-use his example:

``` _("Text with %(num)s placeholders %(symbol)s", num=42, symbol="🧀")```